components/interactive/SearchBar.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+
+import { HashIcon, Search, SearchIcon } from "lucide-react";
+import Image from "next/image";
+import Link from "@/components/links/Link";
+import { useEffect, useState } from "react";
+import { SearchItems } from "@/lib/util/search";
+import Fuse from "fuse.js";
+import SearchItem from "./SearchItem";
+
+export default function SearchBar() {
+  const fuse = new Fuse(SearchItems, {
+    keys: ["name", "url", "description"],
+    threshold: 0.4,
+  });
+
+  const [query, setQuery] = useState("");
+  const [open, setOpen] = useState(false);
+
+  const searchQuery = fuse.search(query).slice(0, 7);
+
+  useEffect(() => {
+    document.addEventListener("keydown", (e) => {
+      if (e.key === "/" && e.ctrlKey) {
+        setOpen(true);
+      }
+
+      if (e.key === "Escape") {
+        setOpen(false);
+      }
+    });
+  }, []);
+
+  return (
+    <div className="not-prose">
+      <button onClick={() => setOpen(true)} className="flex flex-row gap-2">
+        <SearchIcon className="w-8 h-8" />
+        <span className="hidden md:block gap-2 my-auto">
+          Search
+          <span className="ml-4 p-2 outline outline-border rounded-md">
+            Ctrl + /
+          </span>
+        </span>
+      </button>
```

Comment:
Only thing left until approve is making this button look more clickable. I suck at UI soooo @kenos1